parsely/src/main/java/com/parsely/parselyandroid/LocalStorageRepository.kt

```diff
@@ -70,10 +82,9 @@ internal open class LocalStorageRepository(private val context: Context) {
     /**
      * Save the event queue to persistent storage.
      */
-    @Synchronized
-    open fun persistQueue(inMemoryQueue: List<Map<String, Any?>?>) {
-        ParselyTracker.PLog("Persisting event queue")
-        persistObject((inMemoryQueue + getStoredQueue()).distinct())
+    open suspend fun insertEvents(toInsert: List<Map<String, Any?>?>) = mutex.withLock {
```

Comment:
Question (❓): Is the `mutex` usage here and on this repository really needed? 🤔

1. I am asking this because this `insertEvents(...)` is having only one usage, it being called from within the `InMemoryBuffer.init { ... }` block, which is already guarding by another `mutex`. 🤔
2. If that `mutex` is really needed, just because you want to make the repository thread safe (04877191351c14cd440837652fd2eacb0d5d6905), following the same logic, you might want to do the same for the `enqueueEvent(...)` and lock that with a `mutex` as well, no? 🤔
3. IMHO having 2 `mutex` might end-up complicating things a bit, especially if they are not really needed, at least for now. Having said that, I am not sure what are your plans with this repository. Thus, please take this with a gain of salt, I just wanted to raise this question now, even if it is a dump one, instead of ignoring this whole thing altogether. 🙏

Reply:
Good question! I should've left more context here. At this very moment, removing this `mutex` shouldn't change much - the only usage of this method is covered by `mutex` in `InMemoryQueue` already, as you rightly notice (point 1).

But, with next changes, I'm going to make the whole `LocalStorageRepository` thread safe (point 2) and this `mutex` will be needed - it'll make sure that events are not e.g. inserted at the same time when we remove other events. I couldn't add this right away as using `mutex` makes methods `suspend`, making the scope of the PR too big. How does it sound to you?

Reply:
> How does it sound to you?

Sound good @wzieba , it seems that everything is under control and that you have a solid plan going forward, a 👍 from me, thanks for the additional context, much appreciated. 🙇